tests/Parallafka.Tests/CommitPollerTests.cs

```diff
@@ -0,0 +1,141 @@
+            Assert.False(this._poller.CommitNow());
+            Assert.False(this._poller.CommitNow());
+            Assert.False(this._poller.CommitNow());
+            Assert.False(this._poller.CommitNow());
+            Assert.False(this._poller.CommitNow());
+            this._poller.Complete();
+            await this._poller.Completion;
+
+            // then
+            this._committer.Verify(m => m.CommitNow(It.IsAny<CancellationToken>()), Times.Exactly(2));
+        }
+
+        [Fact]
+        public async Task CommitNowCallsCommit()
+        {
+            // when
+            Assert.True(this._poller.CommitNow());
+            this._poller.Complete();
+            await this._poller.Completion;
+
+            // then
+            this._committer.Verify(m => m.CommitNow(It.IsAny<CancellationToken>()), Times.Once);
+        }
+        
+        [Fact]
+        public async Task CommitNowBeforeDelayBothCallCommit()
+        {
+            // when
+            Assert.True(this._poller.CommitNow());
+            Assert.True(this._poller.CommitWithin(TimeSpan.FromSeconds(1)));
+            this._poller.Complete();
+            await this._poller.Completion;
+
+            // then
+            this._committer.Verify(m => m.CommitNow(It.IsAny<CancellationToken>()), Times.Exactly(2));
+        }
+
+        [Fact]
+        public async Task CommitWithinWorks()
+        {
+            // when
+            Assert.True(this._poller.CommitWithin(TimeSpan.FromMilliseconds(100)));
+            await Task.Delay(TimeSpan.FromSeconds(1));
+
+            // then
+            this._committer.Verify(m => m.CommitNow(It.IsAny<CancellationToken>()), Times.Once);
+
+            this._poller.Complete();
+            await this._poller.Completion;
+        }
+
+
+        [Fact]
+        public async Task CommitWithinWithLongDelayIgnoresDelay()
+        {
+            // when
+            var sw = Stopwatch.StartNew();
+            Assert.True(this._poller.CommitWithin(TimeSpan.FromSeconds(100)));
+            this._poller.Complete();
+            await this._poller.Completion;
+
+            // then
+            Assert.True(sw.Elapsed < TimeSpan.FromSeconds(1));
+            this._committer.Verify(m => m.CommitNow(It.IsAny<CancellationToken>()), Times.Once);
+        }
+
+        [Fact]
+        public async Task CommitNowIgnoredAfterCompletion()
+        {
+            // when
+            this._poller.Complete();
+            await this._poller.Completion;
+
+            Assert.False(this._poller.CommitNow());
+
+            // then
+            this._committer.Verify(m => m.CommitNow(It.IsAny<CancellationToken>()), Times.Never);
+        }
+        
+        [Fact]
+        public async Task CommitWithinIgnoredAfterCompletion()
+        {
+            // when
+            this._poller.Complete();
+            await this._poller.Completion;
+
+            Assert.False(this._poller.CommitWithin(TimeSpan.FromSeconds(1)));
+
```

Comment:
Should delay here right?

Reply:
No, the poller is completed (Completion task has finished).  The task is completed, so no new commits can be sent.

Reply:
CommitWithin returns true when the commit was queued, false otherwise.  You want to test that something the method said it didn't do, it won't do?

Reply:
I see your point: the verify should wait for that duration to fully trust the response from CommitWithin.

Reply:
I probably would in this case but I don't feel too strongly about it.
In theory someone could go in and add a `return false if completed` after the enqueue